File: src/main/scala/fundamentals/scala/collections/JavaArrayList.scala
package fundamentals.scala.collections

import java.util
import java.util.ArrayList

import scala.collection.mutable.ArrayBuffer

object JavaArrayList extends App {
  val arrList: util.ArrayList[Int] = new util.ArrayList()
  arrList.add(1)
  arrList.add(2)
  arrList.add(3)
  arrList.add(4)

  var i = 0
  while (i < arrList.size()) {
    println(arrList.get(i))
    i += 1
  }

  // Alternative Scala mutable collection
  val list = ArrayBuffer(1, 2, 3)
  list += 4 // example mutate the list
}

File: src/main/java/fundamentals/java/collections/JavaArrayList.java
package fundamentals.java.collections;

import java.util.ArrayList;

public class JavaArrayList {
    public static void main(String[] args) {
        ArrayList<Integer> list = new ArrayList<>();
        list.add(1);
        list.add(2);
        list.add(3);
        list.add(4);
        for (int i : list) System.out.println(i);
    }
}
